runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -1787,6 +1794,54 @@ TR::Register *J9::X86::TreeEvaluator::multianewArrayEvaluator(TR::Node *node, TR
 
    node->setRegister(targetReg);
    return targetReg;
+}
+
+/**
+ * Generate code for multianewarray
+ *
+ * Checks the number of dimensions. For 1 dimensional arrays call the helper, for >1 call
+ * generateMultianewArrayWithInlineAllocators.
+*/
+TR::Register *J9::X86::TreeEvaluator::multianewArrayEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+
+   TR::Compilation *comp = cg->comp();
+   TR_ASSERT_FATAL(comp->target().is64Bit(), "multianewArrayEvaluator is only supported on 64-bit JVMs!");
+
+   bool useInlineAllocator = false;
+   TR::Node *secondChild = node->getSecondChild();    // Number of dimensions - this is fixed in the bytecode, so compile time constant
+
+   // The number of dimensions should always be an iconst
+   TR_ASSERT_FATAL(secondChild->getOpCodeValue() == TR::iconst, "dims of multianewarray must be iconst");
+
+   // Only generate inline code if nDims > 1
+   uint32_t nDims = secondChild->get32bitIntegralValue();
+   if (nDims > 1)
+      {
+      useInlineAllocator = true;
+      }
+   else
+      {
+      // trace a message to indicate that inline allocation is disabled for nDims < 2
+      if (comp->getOption(TR_TraceCG))
+         {
+         traceMsg(comp, "Disabling inline allocations for multianewarray of dim %d\n", nDims);
+         }
+      }
+
+   if (useInlineAllocator)
+      {
+      return generateMultianewArrayWithInlineAllocators(node, cg);
```

Comment:
Why do you need `useInlineAllocator` at all?  Just move this `return` up to where you set it.

Reply:
It made a little more sense before your last set of requested changes which I implemented a little too literally. I'll clean it up